GameLogic.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace GameOfLife;
+
+use GameOfLife\rules\BaseRule;
+
+class GameLogic
+{
+    private $rule;
+    private $history = [];
+    private $loopDetected = false;
+
+    public function __construct(BaseRule $_rule)
+    {
+        $this->rule = $_rule;
+    }
+
+    public function calculateNextBoard(Board $_board)
+    {
+        for ($y = 0; $y < $_board->height(); $y++)
+        {
+            for ($x = 0; $x < $_board->width(); $x++)
+            {
+                $nextBoard[$y][$x] = 0;
```

Comment:
Doesn't this throw at least PHP notices? I think you should at least initialize `$nextBaord` before using it.